src/server/pfs/fuse/filesystem.go

```diff
@@ -289,7 +297,7 @@ func (h *handle) Write(ctx context.Context, request *fuse.WriteRequest, response
 	// observed on osx, not on linux.
 	repeated := h.written - int(request.Offset)
 	if repeated < 0 {
-		return fmt.Errorf("gap in bytes written, (OpenNonSeekable should make this impossible)")
+		repeated = h.written
```

Comment:
Hmm, I'm not sure that this is correct. What we're basically saying here is that if the offset is past the amount we've written (ie there's a gap) we treat this data as if it's repeating all the data we've already seen, which it probably isn't.

Also I don't think there's any guarantee here that when `repeated < 0` `h.written < len(request.Data)` which it has to be because we use it as an index on `request.Data`.

Under what circumstances do we get `repeated < 0`?